pkg/ipamd/ipamd_test.go

```diff
@@ -2344,3 +2344,84 @@ func TestIsConfigValid(t *testing.T) {
 	}
 
 }
+
+func TestAnnotatePod(t *testing.T) {
+	m := setup(t)
+	defer m.ctrl.Finish()
+	ctx := context.Background()
+
+	// Define the Pod objects to test
+	pod := corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-pod",
+			Namespace: "test-namespace",
+		},
+	}
+
+	mockContext := &IPAMContext{
+		awsClient:       m.awsutils,
+		rawK8SClient:    m.rawK8SClient,
+		cachedK8SClient: m.cachedK8SClient,
+		primaryIP:       make(map[string]string),
+		terminating:     int32(0),
+		networkClient:   m.network,
+		dataStore:       testDatastore(),
+		enableIPv4:      true,
+		enableIPv6:      false,
+	}
+
+	mockContext.rawK8SClient.Create(ctx, &pod)
+
+	ipOne := "10.0.0.1"
+	ipTwo := "10.0.0.2"
+	ipThree := "10.0.0.3"
+
+	//Test basic add operation for new pod
```

Comment:
nit: let's add a space after comment. I am surprised this didn't get done by `go fmt`, but maybe you have an older go installation